### DISCUSSION TITLE
How to fix useSearchParams deopted into client-side rendering?

### DISCUSSION THREAD

**Discussion by muhammadali543** [vercel/next.js#61925] [App Router]
After updating to Next.js 14.1, I get these build messages telling me that all my pages are deopted into client-side rendering due to "useSearchParams().

Here is an example from the terminal when running next build:

Entire page "/404" deopted into client-side rendering due to "useSearchParams()".

https://nextjs.org/docs/messages/deopted-i  (link shows 404 - This page could not be found.)

Please fix this as soon as possible or change useSearchParams to new one. Thanks!

**Comment by icyJoseph** [comment]
Hi,

But, are you using `useSearchParams` in your code? Possibly a dependency?
